snapcraft/meta.py

```diff
@@ -100,13 +100,39 @@ def _setup_assets(meta_dir, config_data):
         _setup_config_hook(hooks_dir, config_data['config'])
 
     if 'license' in config_data:
+        logger.warning("DEPRECATED: 'license' defined in snapcraft.yaml")
         license_path = os.path.join(meta_dir, 'license.txt')
-        shutil.copyfile(config_data['license'], license_path)
+        os.link(config_data['license'], license_path)
 
     if 'icon' in config_data:
+        logger.warning("DEPRECATED: 'icon' defined in snapcraft.yaml")
         icon_ext = config_data['icon'].split(os.path.extsep)[1]
-        icon_path = os.path.join(meta_dir, 'icon.{}'.format(icon_ext))
-        shutil.copyfile(config_data['icon'], icon_path)
+        icon_dir = os.path.join(meta_dir, 'gui')
+        icon_path = os.path.join(icon_dir, 'icon.{}'.format(icon_ext))
+        os.mkdir(icon_dir)
+        os.link(config_data['icon'], icon_path)
+
+    _setup_from_setup(meta_dir)
+
+
+def _setup_from_setup(meta_dir):
```

Comment:
This looks okay but I'd like @mvo5 to ack it as well. I'm not 100% sure of the design of the setup directory.